Autopilot/Path Manager/MPL3115A2.c

```diff
@@ -30,36 +33,80 @@ char initAltimeter() {
 #endif
     }
     return altimeterConnected;
+    // set pressure 
 }
 
-void calibrateAltimeter(long int pressure){
-    int msb = pressure >> 9;
-    int lsb = (pressure >> 1)& 255;
-    int offset = 0;
-    sendMessage(I2C_SLAVE_ADDRESS, BAR_IN_MSB, (char*)&msb, 1, WRITE);
-    sendMessage(I2C_SLAVE_ADDRESS, BAR_IN_LSB, (char*)&lsb, 1, WRITE);
-    sendMessage(I2C_SLAVE_ADDRESS, OFF_H, (char*)&offset, 1, WRITE);
+void calibrateAltimeter(int altitudeOffset){
+    if (altitudeOffset < -128 || altitudeOffset > 128){
+        error("Invalid altitude offset!");
+        // the range of altitude offset should be within -128 to 128
+    }
+    
+    else{
+        switchMode(1); //Barometer mode
+        float pressureFloat = 0.0;
```

Comment:
you only want to do this pressure stuff if the input is 0 for the altmeter offset. Otherwise you should just adjust the altmeter offset register on the altmeter. You should also set the offset to 0 if you do happen to perform the calibration by setting the base pressure as 0